airbyte-integrations/connectors/source-tidb/src/main/java/io/airbyte/integrations/source/tidb/TiDBSourceOperations.java

```diff
@@ -0,0 +1,69 @@
+package io.airbyte.integrations.source.tidb;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.mysql.cj.MysqlType;
+import com.mysql.cj.jdbc.result.ResultSetMetaData;
+import com.mysql.cj.result.Field;
+import io.airbyte.db.SourceOperations;
+import io.airbyte.integrations.source.mysql.MySqlSourceOperations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static io.airbyte.db.jdbc.JdbcConstants.*;
+
+public class TiDBSourceOperations extends MySqlSourceOperations implements SourceOperations<ResultSet, MysqlType> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TiDBSourceOperations.class);
+
+  @Override
+  public void setJsonField(ResultSet resultSet, int colIndex, ObjectNode json) throws SQLException {
+    final ResultSetMetaData metaData = (ResultSetMetaData) resultSet.getMetaData();
+    final Field field = metaData.getFields()[colIndex - 1];
+    final String columnName = field.getName();
+    final MysqlType columnType = field.getMysqlType();
+
+    switch (columnType) {
+      case BIT -> {
+        putBinary(json, columnName, resultSet, colIndex);
+      }
+      default -> {
+        super.setJsonField(resultSet, colIndex, json);
+      }
+    }
+  }
+
+  @Override
+  public MysqlType getFieldType(JsonNode field) {
+    try {
+      final MysqlType literalType = MysqlType.getByName(field.get(INTERNAL_COLUMN_TYPE_NAME).asText());
+      final int columnSize = field.get(INTERNAL_COLUMN_SIZE).asInt();
+
+      switch (literalType) {
+        // TINYINT(1) are interpreted as boolean
+        case TINYINT, TINYINT_UNSIGNED -> {
+          if (columnSize == 1) {
+            return MysqlType.BOOLEAN;
+          }
+        }
+        // When CHAR[N] and VARCHAR[N] columns have binary character set, the returned
+        // types are BINARY[N] and VARBINARY[N], respectively. So we don't need to
+        // convert them here. This is verified in MySqlSourceDatatypeTest.
+      }
+
+      return literalType;
+    } catch (final IllegalArgumentException ex) {
+      LOGGER.warn(String.format("Could not convert column: %s from table: %s.%s with type: %s (type name: %s). Casting to VARCHAR.",
+          field.get(INTERNAL_COLUMN_NAME),
+          field.get(INTERNAL_SCHEMA_NAME),
+          field.get(INTERNAL_TABLE_NAME),
+          field.get(INTERNAL_COLUMN_TYPE),
+          field.get(INTERNAL_COLUMN_TYPE_NAME)));
+      return MysqlType.VARCHAR;
+    }
+  }
```

Comment:
You can remove this function for now, you're not overwritten it. If in the future will be a change let this happens with it, but today the behavior is equal to MySQL.

Reply:
Not exactly same. `bit(1)` is not interpreted as boolean.